In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from funpymodeling.exploratory import freq_tbl

In [7]:
df=pd.read_csv("DF_Mexico.csv",encoding='latin-1')
df

id                                      listing_url  \
0      2.056638e+06             https://www.airbnb.com/rooms/2056638   
1      2.072354e+06             https://www.airbnb.com/rooms/2072354   
2      6.960370e+05              https://www.airbnb.com/rooms/696037   
3      3.579700e+04               https://www.airbnb.com/rooms/35797   
4      4.461600e+04               https://www.airbnb.com/rooms/44616   
...             ...                                              ...   
24219  8.540000e+17  https://www.airbnb.com/rooms/853923589855678923   
24220  8.540000e+17  https://www.airbnb.com/rooms/853966732274073000   
24221  8.540000e+17  https://www.airbnb.com/rooms/854225071295768399   
24222  8.540000e+17  https://www.airbnb.com/rooms/854247693069190865   
24223  8.540000e+17  https://www.airbnb.com/rooms/854260329845513725   

          scrape_id last_scraped       source  \
0      2.020000e+13   30/03/2023  city scrape   
1      2.020000e+13   30/03/2023  city scrape   
2      2.020000e+13   29/03/2023  city scrape   
3      2.020000e+13   29/03/2023  city scrape   
4      2.020000e+13   30/03/2023  city scrape   
...             ...          ...          ...   
24219  2.020000e+13   30/03/2023  city scrape   
24220  2.020000e+13   29/03/2023  city scrape   
24221  2.020000e+13   29/03/2023  city scrape   
24222  2.020000e+13   30/03/2023  city scrape   
24223  2.020000e+13   30/03/2023  city scrape   

                                                    name  \
0                    Amplio y luminoso loft en Coyoacán   
1                     Coyoacan Historic Studio Apartment   
2      3 Bedrooms, 2 blocks from Polanco - ALL RENOVA...   
3                                            Villa Dante   
4                                      CONDESA HAUS  B&B   
...                                                  ...   
24219                              Rincón Mexico lindo!   
24220                     Departamento en zona diamante.   
24221             Habitación privada en magnífica casa   
24222                   Espacioso y luminoso loft - Roma   
24223      Maravilloso departamento en Ciudad de México   

                                             description  \
0      Cómodo loft de dos pisos magníficamente ubic...   
1      This studio flat is adjacent to the owner's ho...   
2      Beautifully decorated 3 bedroom apartment, it ...   
3      Dentro de Villa un estudio de arte con futon, ...   
4      A new concept of hosting in mexico through a b...   
...                                                  ...   
24219  Este lugar tiene una ubicación estratégica: ...   
24220  Ubicado en zona diamante, junto al aeropuerto,...   
24221  Relájate en este espacio muy tranquilo y eleg...   
24222  What else does it come with?<br />- WIFI 100MB...   
24223  Ven a descubrir este increíble departamento e...   

                                   neighborhood_overview  \
0                                                    NaN   
1      Located in Coyoacan, in a quiet neighborhood w...   
2      This area is called Nuevo Polanco, it has bein...   
3      Centro comercial Santa Fe, parque interlomas y...   
4                                                    NaN   
...                                                  ...   
24219                                                NaN   
24220                                                NaN   
24221                                                NaN   
24222                                                NaN   
24223  Una zona muy bonita ya que varias de las atrac...   

                                             picture_url    host_id  ...  \
0      https://a0.muscache.com/pictures/28353712/4379...   10531228  ...   
1      https://a0.muscache.com/pictures/369f3371-593a...   16840050  ...   
2      https://a0.muscache.com/pictures/10960397/67b2...    3531879  ...   
3      https://a0.muscache.com/pictures/f395ab78-1185...     153786  ...   
4      https://a0.mu

**Análisis exploratorio de la base de datos Airbnb**  
1. Dimensión del data frame.
2. Tipos de datos.


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24224 entries, 0 to 24223
Data columns (total 76 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            24224 non-null  float64
 1   listing_url                                   24224 non-null  object 
 2   scrape_id                                     24224 non-null  float64
 3   last_scraped                                  24224 non-null  object 
 4   source                                        24224 non-null  object 
 5   name                                          24223 non-null  object 
 6   description                                   23296 non-null  object 
 7   neighborhood_overview                         15145 non-null  object 
 8   picture_url                                   24224 non-null  object 
 9   host_id                                       24224 non-null 

El DataFrame cuenta con 24,224 observaciones y 75 variables, se aprecia que contamos con tipos de datos numéricos (float e int) y con caracteres (object),
se aprecia que existen variables con datos faltantes.

**Tratamiento de datos nulos**
1. Seleccionamos las variables solicitadas por el profesor.
2. Pre- visualización de los nulos.
3. Tratamiento de datos faltantes justificando la metodología.

In [12]:
valores_nulos=df.isnull().sum()
valores_nulos

id                                                 0
listing_url                                        0
scrape_id                                          0
last_scraped                                       0
source                                             0
                                                ... 
calculated_host_listings_count                     0
calculated_host_listings_count_entire_homes        0
calculated_host_listings_count_private_rooms       0
calculated_host_listings_count_shared_rooms        0
reviews_per_month                               3596
Length: 75, dtype: int64

Dado que en la visualización no podemos observar el total de nulos pasaremos a seleccionar manualmente únicamente las variables que presenten esta condición.

In [10]:
df_seleccion = df.iloc[  : , [5,6,7,13,14,15,16,17,18,21,27,29,35,36,37,38,43,44,45,46,47,48,49,59,60,61,62,63,64,65,66,67,68,74]]
df_seleccion

name  \
0                    Amplio y luminoso loft en Coyoacán   
1                     Coyoacan Historic Studio Apartment   
2      3 Bedrooms, 2 blocks from Polanco - ALL RENOVA...   
3                                            Villa Dante   
4                                      CONDESA HAUS  B&B   
...                                                  ...   
24219                              Rincón Mexico lindo!   
24220                     Departamento en zona diamante.   
24221             Habitación privada en magnífica casa   
24222                   Espacioso y luminoso loft - Roma   
24223      Maravilloso departamento en Ciudad de México   

                                             description  \
0      Cómodo loft de dos pisos magníficamente ubic...   
1      This studio flat is adjacent to the owner's ho...   
2      Beautifully decorated 3 bedroom apartment, it ...   
3      Dentro de Villa un estudio de arte con futon, ...   
4      A new concept of hosting in mexico through a b...   
...                                                  ...   
24219  Este lugar tiene una ubicación estratégica: ...   
24220  Ubicado en zona diamante, junto al aeropuerto,...   
24221  Relájate en este espacio muy tranquilo y eleg...   
24222  What else does it come with?<br />- WIFI 100MB...   
24223  Ven a descubrir este increíble departamento e...   

                                   neighborhood_overview        host_location  \
0                                                    NaN  Mexico City, Mexico   
1      Located in Coyoacan, in a quiet neighborhood w...  Mexico City, Mexico   
2      This area is called Nuevo Polanco, it has bein...  Mexico City, Mexico   
3      Centro comercial Santa Fe, parque interlomas y...  Mexico City, Mexico   
4                                                    NaN  Mexico City, Mexico   
...                                                  ...                  ...   
24219                                                NaN  Mexico City, Mexico   
24220                                                NaN                  NaN   
24221                                                NaN  Mexico City, Mexico   
24222                                                NaN  Mexico City, Mexico   
24223  Una zona muy bonita ya que varias de las atrac...                  NaN   

                                              host_about host_response_time  \
0                                                    NaN       within a day   
1                                                    NaN                NaN   
2      We are from CDMX and moved back and forth to N...     within an hour   
3      Master in visual arts, film photography & Mark...                NaN   
4      Condesa Haus  is Bed & Breakfast located on th...     within an hour   
...                                                  ...                ...   
24219  Amo viajar y disfrutar de mis viajes y que mej...     within an hour   
24220                                                NaN                NaN   
24221  Me considero una persona accesible para mis hu...                NaN   
24222                                                NaN     within an hour   
24223                                                NaN                NaN   

       host_response_rate  Unnamed: 17  host_acceptance_rate  \
0                     1.0          NaN                   NaN   
1                     NaN          NaN                   NaN   
2                     1.0          NaN                  0.90   
3                     NaN          NaN                   NaN   
4                     1.0          NaN                  0.86   
...                   ...          ...                   ...   
24219                 1.0          NaN                  1.00   
24220                 NaN          NaN                   NaN   
24221                 NaN          NaN                  0.96   
24222                 1.0          NaN          

In [16]:
valores_nulos=df_seleccion.isnull().sum()
valores_nulos

name                                               1
description                                      928
neighborhood_overview                           9079
host_location                                   5172
host_about                                     10664
host_response_time                              3091
host_response_rate                              3091
Unnamed: 17                                    24224
host_acceptance_rate                            2043
host_picture_url                                   0
host_identity_verified                             0
neighbourhood_cleansed                             0
accommodates                                       0
bathrooms                                      24224
bathrooms_text                                    30
bedrooms                                         555
maximum_nights                                     0
minimum_minimum_nights                             1
maximum_minimum_nights                        

Guardamos en una base los datos que no necesitan tratamiento de datos faltantes.

In [14]:
df_NoNull = df.iloc[  : , [0,1,2,3,4,8,9,10,11,12,19,20,22,23,24,25,26,28,30,31,32,33,34,39,40,41,42,50,51,52,53,54,55,56,57,58,69,70,71,72,73,75] ]
df_NoNull

id                                      listing_url  \
0      2.056638e+06             https://www.airbnb.com/rooms/2056638   
1      2.072354e+06             https://www.airbnb.com/rooms/2072354   
2      6.960370e+05              https://www.airbnb.com/rooms/696037   
3      3.579700e+04               https://www.airbnb.com/rooms/35797   
4      4.461600e+04               https://www.airbnb.com/rooms/44616   
...             ...                                              ...   
24219  8.540000e+17  https://www.airbnb.com/rooms/853923589855678923   
24220  8.540000e+17  https://www.airbnb.com/rooms/853966732274073000   
24221  8.540000e+17  https://www.airbnb.com/rooms/854225071295768399   
24222  8.540000e+17  https://www.airbnb.com/rooms/854247693069190865   
24223  8.540000e+17  https://www.airbnb.com/rooms/854260329845513725   

          scrape_id last_scraped       source  \
0      2.020000e+13   30/03/2023  city scrape   
1      2.020000e+13   30/03/2023  city scrape   
2      2.020000e+13   29/03/2023  city scrape   
3      2.020000e+13   29/03/2023  city scrape   
4      2.020000e+13   30/03/2023  city scrape   
...             ...          ...          ...   
24219  2.020000e+13   30/03/2023  city scrape   
24220  2.020000e+13   29/03/2023  city scrape   
24221  2.020000e+13   29/03/2023  city scrape   
24222  2.020000e+13   30/03/2023  city scrape   
24223  2.020000e+13   30/03/2023  city scrape   

                                             picture_url    host_id  \
0      https://a0.muscache.com/pictures/28353712/4379...   10531228   
1      https://a0.muscache.com/pictures/369f3371-593a...   16840050   
2      https://a0.muscache.com/pictures/10960397/67b2...    3531879   
3      https://a0.muscache.com/pictures/f395ab78-1185...     153786   
4      https://a0.muscache.com/pictures/251410/ec75fe...     196253   
...                                                  ...        ...   
24219  https://a0.muscache.com/pictures/miso/Hosting-...  488030249   
24220  https://a0.muscache.com/pictures/miso/Hosting-...  506816029   
24221  https://a0.muscache.com/pictures/db71951a-8696...  154670560   
24222  https://a0.muscache.com/pictures/miso/Hosting-...    7799414   
24223  https://a0.muscache.com/pictures/miso/Hosting-...  506864314   

                                          host_url  \
0       https://www.airbnb.com/users/show/10531228   
1       https://www.airbnb.com/users/show/16840050   
2        https://www.airbnb.com/users/show/3531879   
3         https://www.airbnb.com/users/show/153786   
4         https://www.airbnb.com/users/show/196253   
...                                            ...   
24219  https://www.airbnb.com/users/show/488030249   
24220  https://www.airbnb.com/users/show/506816029   
24221  https://www.airbnb.com/users/show/154670560   
24222    https://www.airbnb.com/users/show/7799414   
24223  https://www.airbnb.com/users/show/506864314   

                                host_name  host_since  ... availability_365  \
0                                   Maria  09/12/2013  ...              324   
1                                 Mónica  16/06/2014  ...              346   
2                        Gonzalo & Sandra  10/09/2012  ...               87   
3                                    Dici  28/06/2010  ...              363   
4      Condesa Haus Bed & Breakfast  CDMX  09/08/2010  ...              357   
...                                   ...         ...  ...              ...   
24219                                 Ana  16/11/2022  ...              352   
24220                   Jacqueline Vivian  24/03/2023  ...              365   
24221                                Aroa  15/10/2017  ...              180   
24222                               Tiane  29/07/2013  ...              344   
24223                         Empresarial  24/03/2023  ...              359   

      calendar_last_scraped number_of_reviews  number_of_reviews_ltm  license  \
0                30/03/2023    

Procedemos a eliminar las variables que no contengan datos

In [15]:
df_seleccion2 = df.iloc[  : , [5,6,7,13,14,15,16,17,18,21,27,36,37,38,43,44,45,46,47,48,59,60,61,62,63,64,65,66,67,74]]
df_seleccion2

name  \
0                    Amplio y luminoso loft en Coyoacán   
1                     Coyoacan Historic Studio Apartment   
2      3 Bedrooms, 2 blocks from Polanco - ALL RENOVA...   
3                                            Villa Dante   
4                                      CONDESA HAUS  B&B   
...                                                  ...   
24219                              Rincón Mexico lindo!   
24220                     Departamento en zona diamante.   
24221             Habitación privada en magnífica casa   
24222                   Espacioso y luminoso loft - Roma   
24223      Maravilloso departamento en Ciudad de México   

                                             description  \
0      Cómodo loft de dos pisos magníficamente ubic...   
1      This studio flat is adjacent to the owner's ho...   
2      Beautifully decorated 3 bedroom apartment, it ...   
3      Dentro de Villa un estudio de arte con futon, ...   
4      A new concept of hosting in mexico through a b...   
...                                                  ...   
24219  Este lugar tiene una ubicación estratégica: ...   
24220  Ubicado en zona diamante, junto al aeropuerto,...   
24221  Relájate en este espacio muy tranquilo y eleg...   
24222  What else does it come with?<br />- WIFI 100MB...   
24223  Ven a descubrir este increíble departamento e...   

                                   neighborhood_overview        host_location  \
0                                                    NaN  Mexico City, Mexico   
1      Located in Coyoacan, in a quiet neighborhood w...  Mexico City, Mexico   
2      This area is called Nuevo Polanco, it has bein...  Mexico City, Mexico   
3      Centro comercial Santa Fe, parque interlomas y...  Mexico City, Mexico   
4                                                    NaN  Mexico City, Mexico   
...                                                  ...                  ...   
24219                                                NaN  Mexico City, Mexico   
24220                                                NaN                  NaN   
24221                                                NaN  Mexico City, Mexico   
24222                                                NaN  Mexico City, Mexico   
24223  Una zona muy bonita ya que varias de las atrac...                  NaN   

                                              host_about host_response_time  \
0                                                    NaN       within a day   
1                                                    NaN                NaN   
2      We are from CDMX and moved back and forth to N...     within an hour   
3      Master in visual arts, film photography & Mark...                NaN   
4      Condesa Haus  is Bed & Breakfast located on th...     within an hour   
...                                                  ...                ...   
24219  Amo viajar y disfrutar de mis viajes y que mej...     within an hour   
24220                                                NaN                NaN   
24221  Me considero una persona accesible para mis hu...                NaN   
24222                                                NaN     within an hour   
24223                                                NaN                NaN   

       host_response_rate  Unnamed: 17  host_acceptance_rate  \
0                     1.0          NaN                   NaN   
1                     NaN          NaN                   NaN   
2                     1.0          NaN                  0.90   
3                     NaN          NaN                   NaN   
4                     1.0          NaN                  0.86   
...                   ...          ...                   ...   
24219                 1.0          NaN                  1.00   
24220                 NaN          NaN                   NaN   
24221                 NaN          NaN                  0.96   
24222                 1.0          NaN          

In [16]:
valores_nulos=df_seleccion.isnull().sum()
valores_nulos

name                               1
description                      928
neighborhood_overview           9079
host_location                   5172
host_about                     10664
host_response_time              3091
host_response_rate              3091
host_acceptance_rate            2043
host_is_superhost                  5
host_neighbourhood             12401
neighbourhood                   9079
bathrooms_text                    30
bedrooms                         555
beds                             271
minimum_minimum_nights             1
maximum_minimum_nights             1
minimum_maximum_nights             1
maximum_maximum_nights             1
minimum_nights_avg_ntm             1
maximum_nights_avg_ntm             1
first_review                    3596
last_review                     3596
review_scores_rating            3596
review_scores_accuracy          3652
review_scores_cleanliness       3651
review_scores_checkin           3652
review_scores_communication     3652
r

Vamos a separar 3 bases de datos, cada una representará cada tipo de dato.

In [25]:
df_seleccion2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24224 entries, 0 to 24223
Data columns (total 30 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   name                         24223 non-null  object 
 1   description                  23296 non-null  object 
 2   neighborhood_overview        15145 non-null  object 
 3   host_location                19052 non-null  object 
 4   host_about                   13560 non-null  object 
 5   host_response_time           21133 non-null  object 
 6   host_response_rate           21133 non-null  object 
 7   host_acceptance_rate         22181 non-null  object 
 8   host_is_superhost            24219 non-null  object 
 9   host_neighbourhood           11823 non-null  object 
 10  neighbourhood                15145 non-null  object 
 11  bathrooms_text               24194 non-null  object 
 12  bedrooms                     23669 non-null  float64
 13  beds            

**Object**

In [ ]:
object_df=df_seleccion2.iloc[  : , [0,1,2,3,4,5,6,7,8,9,10,11,20,21]]
object_df

In [37]:
#Reemplazamos datos nulos utilizando BFill
object_df2=object_df.fillna(method="bfill") 
object_df2.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24224 entries, 0 to 24223
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   name                   24224 non-null  object
 1   description            24224 non-null  object
 2   neighborhood_overview  24224 non-null  object
 3   host_location          24223 non-null  object
 4   host_about             24222 non-null  object
 5   host_response_time     24223 non-null  object
 6   host_response_rate     24223 non-null  object
 7   host_acceptance_rate   24224 non-null  object
 8   host_is_superhost      24224 non-null  object
 9   host_neighbourhood     24209 non-null  object
 10  neighbourhood          24224 non-null  object
 11  bathrooms_text         24224 non-null  object
 12  first_review           24189 non-null  object
 13  last_review            24189 non-null  object
dtypes: object(14)
memory usage: 2.6+ MB


In [38]:
#Completamos usando FFill
object_df3=object_df2.fillna(method="ffill") 
object_df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24224 entries, 0 to 24223
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   name                   24224 non-null  object
 1   description            24224 non-null  object
 2   neighborhood_overview  24224 non-null  object
 3   host_location          24224 non-null  object
 4   host_about             24224 non-null  object
 5   host_response_time     24224 non-null  object
 6   host_response_rate     24224 non-null  object
 7   host_acceptance_rate   24224 non-null  object
 8   host_is_superhost      24224 non-null  object
 9   host_neighbourhood     24224 non-null  object
 10  neighbourhood          24224 non-null  object
 11  bathrooms_text         24224 non-null  object
 12  first_review           24224 non-null  object
 13  last_review            24224 non-null  object
dtypes: object(14)
memory usage: 2.6+ MB


Validamos que los datos solicitados sean todos caracteres

In [43]:
freq_tbl(object_df3)

                                                 name  frequency  percentage  \
0                     Perfecto Loft en gran ubicación         48    0.001982   
1              Single and shared rooms near downtown.         24    0.000991   
2                              Departamento 2 cuartos         20    0.000826   
3                                 Hermosa suite doble         13    0.000537   
4      Sky Roma - Suites & Depas, Colonia Roma Norte.         12    0.000495   
...                                               ...        ...         ...   
23341             Habitación en departamento chevere!          1    0.000041   
23342       "Vive Mexico" en el corazon de Roma Norte          1    0.000041   
23343                   Renta recamara para señoritas          1    0.000041   
23344               Luxury and Spacious 3BD apartment          1    0.000041   
23345    Maravilloso departamento en Ciudad de México          1    0.000041   

       cumulative_perc  
0             

Las variables "host_response_rate" y "host_acceptance_rate" parecen ser de tipo float por lo que las enviaremos al DB correspondiente.

**Float**

In [34]:
float_df=df_seleccion.iloc[ :, [12,13,14,15,16,17,18,19,22,23,24,25,26,27,28,29]]
float_df

bedrooms  beds  minimum_minimum_nights  maximum_minimum_nights  \
0           1.0   1.0                     1.0                     1.0   
1           1.0   1.0                     3.0                     3.0   
2           3.0   3.0                    24.0                    24.0   
3           1.0   1.0                     1.0                     1.0   
4           5.0   8.0                     1.0                     1.0   
...         ...   ...                     ...                     ...   
24219       1.0   1.0                     3.0                     3.0   
24220       2.0   3.0                     1.0                     1.0   
24221       2.0   1.0                     5.0                     5.0   
24222       1.0   1.0                     1.0                     1.0   
24223       3.0   3.0                     1.0                     1.0   

       minimum_maximum_nights  maximum_maximum_nights  minimum_nights_avg_ntm  \
0                      1125.0                  1125.0                     1.0   
1                      1125.0                  1125.0                     3.0   
2                      1125.0                  1125.0                    24.0   
3                         7.0                     7.0                     1.0   
4                       365.0                   365.0                     1.0   
...                       ...                     ...                     ...   
24219                  1125.0                  1125.0                     3.0   
24220                   365.0                   365.0                     1.0   
24221                   365.0                   365.0                     5.0   
24222                   365.0                   365.0                     1.0   
24223                  1125.0                  1125.0                     1.0   

       maximum_nights_avg_ntm  review_scores_rating  review_scores_accuracy  \
0                      1125.0                  4.95                    4.71   
1                      1125.0                  4.85                    4.90   
2                      1125.0                  4.57                    4.82   
3                         7.0                   NaN                     NaN   
4                       365.0                  4.58                    4.56   
...                       ...                   ...                     ...   
24219                  1125.0                   NaN                     NaN   
24220                   365.0                   NaN                     NaN   
24221                   365.0                   NaN                     NaN   
24222                   365.0                   NaN                     NaN   
24223                  1125.0                   NaN                     NaN   

       review_scores_cleanliness  review_scores_checkin  \
0                           4.95                   5.00   
1                           4.88                   4.93   
2                           4.71                   4.92   
3                            NaN                    NaN   
4                           4.70                   4.87   
...                          ...                    ...   
24219                        NaN                    NaN   
24220                        NaN                    NaN   
24221                        NaN                    NaN   
24222                        NaN                    NaN   
24223                        NaN                    NaN   

       review_scores_communication  review_scores_location  \
0                             4.95                    5.00   
1                             5.00                    4.95   
2                             4.92                    4.58   
3                              NaN                     NaN   
4                             4.78                    4.98   
...                            ...                     ...   
24219                          NaN                     NaN   
242

In [39]:
#Reemplazamos los datos utilizando la media
float_df2=float_df.fillna(round(float_df.mean(),1))
float_df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24224 entries, 0 to 24223
Data columns (total 16 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   bedrooms                     24224 non-null  float64
 1   beds                         24224 non-null  float64
 2   minimum_minimum_nights       24224 non-null  float64
 3   maximum_minimum_nights       24224 non-null  float64
 4   minimum_maximum_nights       24224 non-null  float64
 5   maximum_maximum_nights       24224 non-null  float64
 6   minimum_nights_avg_ntm       24224 non-null  float64
 7   maximum_nights_avg_ntm       24224 non-null  float64
 8   review_scores_rating         24224 non-null  float64
 9   review_scores_accuracy       24224 non-null  float64
 10  review_scores_cleanliness    24224 non-null  float64
 11  review_scores_checkin        24224 non-null  float64
 12  review_scores_communication  24224 non-null  float64
 13  review_scores_lo